### PR TITLE
[ioctl] fix err revert nil

### PR DESCRIPTION
**Pull request by huof6829**
# Description
prevent err change to nil after run `f.Close`

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Liuhaai — APPROVED** [review]

File: ioctl/doc/doc.go [hunk 1]
```diff
@@ -38,6 +38,9 @@ func GenMarkdownTreeCustom(c *cobra.Command, dir string, name string, path strin
 		return err
 	}
 	defer func() {
+		if err != nil {
+			return
+		}
```
Comment (dustinxie) [inline ioctl/doc/doc.go:43]: if err != nil (for instance, write sth into file failed), we still need to call `f.Close()`

Comment (huof6829) [reply ioctl/doc/doc.go:43]: fixed

Comment (Liuhaai) [reply ioctl/doc/doc.go:43]: f.close()
return err

Comment (huof6829) [reply ioctl/doc/doc.go:43]: f.close() should handle return error.
it's same with return f.close()

Comment (dustinxie) [reply ioctl/doc/doc.go:43]:
```
defer func()
err1 := f.Close()
if err == nil {
    err = err1
}
```

Comment (huof6829) [reply ioctl/doc/doc.go:44]: 👍

**Review by dustinxie — APPROVED** [review]

**Pull request merged**